commands/dht_daemon_test.go

```diff
@@ -0,0 +1,50 @@
+package commands_test
+
+import (
+	"testing"
+	"time"
+	//	"fmt"
+
+	"gx/ipfs/QmR8BauakNcBa3RbE4nbQu76PDiJgoQgz8AJdhJuiU4TAw/go-cid"
+	mh "gx/ipfs/QmerPMzPk1mJVowm8KgmoknWa4yCYvvugMPsgWmDNUvDLW/go-multihash"
+	"gx/ipfs/QmPVkJMTeRC6iBByPWdrRkD3BE5UXsj5HPzb4kPqL186mS/testify/assert"
+	"gx/ipfs/QmPVkJMTeRC6iBByPWdrRkD3BE5UXsj5HPzb4kPqL186mS/testify/require"
+
+	th "github.com/filecoin-project/go-filecoin/testhelpers"
+)
+
+const relayRendevous = "/libp2p/relay"
+
+// Taken from https://github.com/libp2p/go-libp2p-discovery/blob/master/routing.go
+// This function takes a name space string and returns the cid used as a
+// DHT key by providers of the service given by the namespace.
+func nsToCid(ns string) (cid.Cid, error) {
+	h, err := mh.Sum([]byte(ns), mh.SHA2_256, -1)
+	if err != nil {
+		return cid.Undef, err
+	}
+
+	return cid.NewCidV1(cid.Raw, h), nil
+}
+
+func TestSwarmSeesRelayProvider(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	assert := assert.New(t)
+
+	dClient := th.NewDaemon(t, th.SwarmAddr("/ip4/0.0.0.0/tcp/6000")).Start()
+	defer dClient.ShutdownSuccess()
+
+	dProvider := th.NewDaemon(t, th.SwarmAddr("/ip4/0.0.0.0/tcp/6001"), th.IsRelay).Start()
+	defer dProvider.ShutdownSuccess()
+
+	dClient.ConnectSuccess(dProvider)
+
+	time.Sleep(30 * time.Second)
```

Comment:
Any lower and this consistently fails.  It seems important to get some coverage on relay and dht command.  Do we think introducing this potential flake / test delay is worth it?

Reply:
I would say no, or at least not in the same class of tests as other daemon tests. Assuming it's not possible to fake out libp2p to exercise this faster 😢 , I would put this in a silo of long-running integration tests that are *not* part of normal developer workflows: maybe periodically on CI like the functional tests.